src/CommonDBChild.php

```diff
@@ -440,6 +440,11 @@ public function prepareInputForAdd($input)
             static::$mustBeAttached
             && !$this->getItemFromArray(static::$itemtype, static::$items_id, $input)
         ) {
+            Session::addMessageAfterRedirect(
```

Comment:
While this seems indeed a good idea, maybe we should rely on something more similar than `checkAttachedItemChangesAllowed()` do.

Reply:
I am a bit afraid with the addition of rights checks here. It may break back-end code (e.g. crons, console commands, hardcoded workflows, ...) that are expected to be aexecuted with our without any user session.

The idea here was just to add a message to prevent silent creation failures that are related to an invalid input.